apps/web/app/modules/Admin/EditCourse/CourseStatus/CourseStatus.tsx

```diff
@@ -0,0 +1,109 @@
+import * as RadioGroup from "@radix-ui/react-radio-group";
+
+import { Button } from "~/components/ui/button";
+import { Card, CardContent, CardHeader } from "~/components/ui/card";
+import { Form, FormField, FormItem, FormMessage } from "~/components/ui/form";
+import { cn } from "~/lib/utils";
+
+import { useCourseStatusForm } from "./hooks/useCourseStatusForm";
+
+type CourseStateProps = {
+  courseId: string;
+  state?: string;
+};
+
+const CourseState = ({ courseId, state }: CourseStateProps) => {
+  const { form, onSubmit } = useCourseStatusForm({ courseId, state });
+
+  return (
+    <div className="w-full flex gap-4">
+      <Card className="w-full max-w-3xl">
+        <CardHeader>
+          <h5 className="text-xl font-semibold">State</h5>
+        </CardHeader>
+        <CardContent>
+          <p>Set state for the course</p>
+          <Form {...form}>
+            <form className="mt-10 space-y-6" onSubmit={form.handleSubmit(onSubmit)}>
+              <FormField
+                name="state"
+                control={form.control}
+                render={({ field }) => (
+                  <FormItem>
+                    <RadioGroup.Root
+                      {...field}
+                      className="flex flex-col space-y-4"
+                      value={field.value}
+                      onValueChange={field.onChange}
+                    >
+                      <RadioGroup.Item
+                        value="draft"
+                        id="draft"
+                        className={cn(
+                          `p-4 border rounded-md cursor-pointer text-left ${
+                            field.value === "draft" ? "border-blue-500" : "border-gray-300"
+                          }`,
```

Comment:
**suggestion**: The `cn` function is not used correctly here. 
It's meant to simplify conditional class merging, so you can refactor this to avoid embedding template literals directly. 
Consider passing conditions as separate arguments.
```suggestion
                        className={cn(
                          "p-4 border rounded-md cursor-pointer text-left", {
                            "border-blue-500": field.value === "draft",  
                            "border-gray-300": field.value !== "draft",  
                          },
```